### DISCUSSION TITLE
reference to <some effect> is Ambiguous

### DISCUSSION THREAD

**Discussion by weow147** [pschatzmann/arduino-audio-tools#1253] [Q&A] [answered]
I'd like to use AudioEffects and AudioSTK simultaneously, but it looks like both of these libraries are trying to define the same name (Delay).

I'm not aware of any way to force the compiler to use the audioeffects library (instead of audiostk) in the constructor?   Is there a workaround?

```
#include "AudioEffects/AudioEffects.h"
#include "AudioLibs/AudioSTK.h" //cant use both at the same time for DELAY
Delay dlay(1000, 0.5, 1.0, 44100, false);
```
----
`Compilation error: reference to 'Delay' is ambiguous`

**Comment by pschatzmann** [comment, marked as answer]
That's why we have namespaces in c++: Just indicate the namespace that you want and you will be fine:
e.g. audio_tools::Delay dlay(1000, 0.5, 1.0, 44100);

**Reply by weow147** [reply]
I'm more accustomed to navigating down into classes with dot and intellisense... 
I tried to get there with:
AudioEffects.AudioEffect.Delay dlay(1000, 0.5, 1.0, 44100);
...didnt work...
not overly familiar with the :: operator, i'll do some reading.

for namespaces i'm more familiar with 'using namespaceXYZ'

anyway THANKYOU for the assist, i would not have gotten there on my own.

**Comment by pschatzmann** [comment]
Arduino libraries (usually) do not use namespaces, which can lead to conflicts like this.

As a way around this, I started to use namespaces in my libraries and I have added the "using namespace" (which is by default active) in the global include, though this is considered to be an antipattern in C++ programming.
